.storybook/decorators/PropCombinations/README.md

```diff
@@ -0,0 +1,275 @@
+
+```tsx
+const meta: StoryMeta<typeof Button> = {
+  ...
+  parameters: {
+    generate: {
+      stories: ['LargeSize', 'DefaultSize'],
+      props: { ... },
+    },
+  }
+}
+
+...
+
+export const DefaultSize: StoryType<typeof Button> = () => <></>;
+DefaultSize.parameters = {
+  generate: {
+    args: {
+      size: Size.Default,
+    },
+  },
+};
+
+export const LargeSize: StoryType<typeof Button> = () => <></>;
+LargeSize.parameters = {
+  generate: {
+    args: {
+      size: Size.Large,
+    },
+  },
+};
+```
+
+# Options
+
+## `args`
+
+`Record<key in keyof ComponentProps, ComponentProps[key]>`
+
+By default the decorator pulls args from the main `meta.args` value. However, you may need to set args that are different for the generated instances. In this case,
+use `generate.args` to pass static props that are different from the main story args.
+
+```tsx
+const meta: StoryMeta<typeof Tooltip> = {
+  component: Tooltip,
+  parameters: {
+    default: 'LiveExample',
+    generate: {
+      props: { ... },
+      args: {open: true}
+    },
+  },
+};
+export default meta;
+```
+
+## `decorator`
+
+`(InstanceFn: StoryFn, context?: {args: Args}) => JSX.Element`
+
+The GeneratedStory decorator will _not_ read the decorators value defined in `meta.decorators`. To add a wrapper around each instance, define a decorator within `generate`.
+
+```tsx
+const meta: StoryMeta<typeof Body> = {
+  component: Body,
+  parameters: {
+    default: 'LiveExample',
+    generate: {
+      props: { ... },
+      decorator: (InstanceFn, context) => {
+        return (
+          <LeafyGreenProvider baseFontSize={context?.args.baseFontSize}>
+            <InstanceFn />
+          </LeafyGreenProvider>
+        );
+      },
+    },
+  },
+};
+```
+
+## `excludeCombinations`
+
+```ts
+Array<
+  // Mutually exclusive props
+  [keyof ComponentProps, keyof ComponentProps] |
+  // Conditional props
+  [keyof ComponentProps, Record<key in keyof ComponentProps, ComponentProps[key]>]
+  // Specific combinations
+  Record<key in keyof ComponentProps, ComponentProps[key]> |
+>
+```
+
+You may not always want to test _every_ combination of props. For example, in Button we shouldn't be testing the case when `children`, `leftGlyph` and `rightGlyph` are _all_ undefined.
+
+To exclude prop combinations, add `excludeCombinations` to the `generate` object:
+
+```tsx
+generate: {
+  props: { ... }
+  excludeCombinations: [
+    {
+      children: undefined,
+      rightGlyph: undefined,
+      leftGlyph: undefined,
+    },
+  ],
+}
+```
+
+There are three methods of defining `excludeCombinations`.
+
+### Mutually Exclusive combinations
+
+Mutually exclusive combinations check that only one of the defined props exist on the instance. If they are both defined, then the instance is ignored.
+
+For example, a component might accept a `children` prop, or a `title` prop, but never both.
+
+Define mutually exclusive combinations as a tuple of prop keys
```

Comment:
Since this is defined as a tuple instead of an array, that implies we'll never want to define a set of 3 mutually-exclusive props here, which I don't see a whole lot of reason to prevent that here (though ideally we shouldn't have that case much).

Additionally, tuples are usually an ordered pair, while this doesn't appear to care about the order.

Reply:
It's not strictly a tuple, no, just an array of length 2.

I never came across a situation where more than 2 props were mutually exclusive. We could update the function if we ever need this

Reply:
Maybe `array of two elements` would be clearer here. `tuple` is a very specific data type, so I agree it can be confusing.